app/models/direct_award.py

```diff
@@ -81,12 +100,15 @@ class DirectAwardSearch(db.Model):
     archived_services = db.relationship(ArchivedService, secondary='direct_award_search_result_entries',
                                         order_by=lambda: DirectAwardSearchResultEntry.id, lazy='dynamic')
 
-    project = db.relationship(DirectAwardProject)
+    project = db.relationship(DirectAwardProject, backref="searches")
     user = db.relationship(User)
 
-    # Partial index on project_id,active==1. Enforces only one active search per project at a time.
-    __table_args__ = (db.Index('idx_project_id_active', project_id, active, unique=True,
-                               postgresql_where=db.Column('active')),)
+    __table_args__ = (
+        # this may appear tautological, but is required to be able to make a compound foreign key to these together
+        db.UniqueConstraint(id, project_id, name="uq_direct_award_searches_id_project_id"),
+        # Partial index on project_id,active==1. Enforces only one active search per project at a time.
+        db.Index('idx_project_id_active', project_id, active, unique=True, postgresql_where=db.Column('active')),
```

Comment:
This is very cool but the comment could be clearer. Also we can use the code to be more descriptive with:

`db.Index('idx_project_id_active', project_id, active, unique=True, postgresql_where=(db.Column('active') == sql_true()))`

Something along the lines of:
`# Create a partial index on `project_id` and `active == True` to enforce a maximum of one active search per project`

Reply:
oh ok i was going for brevity but yeah